flang/lib/Parser/executable-parsers.cpp

```diff
@@ -282,18 +281,26 @@ TYPE_CONTEXT_PARSER("loop control"_en_US,
                 "CONCURRENT" >> concurrentHeader,
                 many(Parser<LocalitySpec>{})))))
 
+// "DO" is a valid statement, so the loop control is optional; but for
+// better recovery from errors in the loop control, don't parse a
+// DO statement with a bad loop control as a DO statement that has
+// no loop control and is followed by garbage.
+static constexpr auto loopControlOrEndOfStmt{
+    construct<std::optional<LoopControl>>(Parser<LoopControl>{}) ||
+    lookAhead(":\n"_ch) >> construct<std::optional<LoopControl>>()};
```

Comment:
So this is trying to construct parsed loop control, and failing that it'll construct an empty loop control.
Could you please help me understand `lookAhead(":\n"_ch)`? Why is there `:` and what's the significance of user defined literal `_ch`?

Reply:
Ok, I figured out `_ch` part, but still have a question about why `:` is used in `lookAhead()`

Reply:
The ':' (colon) is a typo, and should be a ';' semicolon; will fix.  It's checking for the end of the statement.